pageserver/src/tenant/remote_timeline_client/download.rs

```diff
@@ -149,10 +148,16 @@ pub async fn download_layer_file<'a>(
         .with_context(|| format!("rename download layer file to {local_path}"))
         .map_err(DownloadError::Other)?;
 
-    crashsafe::fsync_async(&local_path)
+    // We use fatal_err() below because the after the rename above,
+    // the in-memory state of the filesystem already has the layer file in its final place,
+    // and subsequent pageserver code could think it's durable while it really isn't.
+    let timeline_dir = VirtualFile::open(&timeline_path)
         .await
-        .with_context(|| format!("fsync layer file {local_path}"))
-        .map_err(DownloadError::Other)?;
+        .fatal_err("VirtualFile::open for timeline dir fsync");
+    timeline_dir
+        .sync_all()
+        .await
+        .fatal_err("VirtualFile::sync_all timeline dir");
```

Comment:
Hm, I was going to argue that we will this PR after #6378 , so, the VirtualFile::sync_all() will be asynchronous on tokio-epoll-uring. 

But, with `std-fs`, it will block the executor thread.
So, would need to bifurcate this.

Reply:
I pushed https://github.com/neondatabase/neon/pull/6991/commits/ca616ee10bfd95104fff51f0a195966153e8b9ff to add the bifurcation.

Yet, before #6378  is merged, which makes `sync_all()` truly async for tokio-epoll-uring, we shouldn't merge this PR.